pandas/core/frame.py

```diff
@@ -912,6 +916,17 @@ def __init__(
 
         NDFrame.__init__(self, mgr)
 
+        if original_dtype is None and is_pandas_object and data_dtype == np.object_:
+            if self.dtypes.iloc[0] != data_dtype:
+                warnings.warn(
+                    "Dtype inference on a pandas object "
```

Comment:
"in the DataFrame constructor" somewhere in this sentence?  I think the "The Index constructor" below is a copy/paste leftover

Reply:
yeah sorry, adjusted